types/chartist/chartist-tests.ts

```diff
@@ -563,4 +543,247 @@ new Chartist.Line('.ct-chart', {
+    new Chartist.Bar(
+        '.bar',
+        {
+            series: [],
+            labels: [],
+        },
+        {
+            plugins: [Chartist.plugins.superCharts()],
+        }
+    )
+);
+
+// -- SVG API
+
+var svgNode: SVGGElement = document.createElementNS('http://www.w3.org/2000/svg', 'g');
+var chartistSvgGroup: Chartist.IChartistSvg<SVGGElement> = new Chartist.Svg(svgNode);
+var charistSvgLine: Chartist.IChartistSvg<SVGLineElement> | null = chartistSvgGroup.querySelector('line');
+
+// --- Axis API
+
+var customStepAxisX: Chartist.IChartistStepAxis<Chartist.ChartistAxisUnitX> = {
+    units: {pos: "x", len: "width", dir: "horizontal", rectStart: "x1", rectEnd: "x2", rectOffset: "y2"},
+    counterUnits: {pos: "y", len: "height", dir: "vertical", rectStart: "y2", rectEnd: "y1", rectOffset: "x1"},
+    chartRect: {
+        padding: {top: 12, right: 16, bottom: 0, left: 0},
+        y2: 12,
+        y1: 120,
+        x1: 40,
+        x2: 746,
+        width: () => 0,
+        height: () => 0,
+    },
+    axisLength: 706,
+    gridOffset: 12,
+    ticks: ["Jul '18", "Aug '18", "Sep '18", "Oct '18", "Nov '18", "Dec '18", "Jan '19", "Feb '19", "Mar '19", "Apr '19", "May '19"],
+    options: {
+        offset: 30,
+        position: "end",
+        labelOffset: {x: 0, y: 0},
+        showLabel: true,
+        showGrid: true,
+        ticks: ["Jul '18", "Aug '18", "Sep '18", "Oct '18", "Nov '18", "Dec '18", "Jan '19", "Feb '19", "Mar '19", "Apr '19", "May '19"],
+        stretch: false,
+    },
+    stepLength: 64.18181818181819,
+    projectValue: (a: number) => a,
+};
+
+var axisX: Chartist.IChartistAxis<Chartist.ChartistAxisUnitX> = customStepAxisX;
+
+// $ExpectError
+var invalidAxisY: Chartist.IChartistAxis<Chartist.ChartistAxisUnitY> = axisX;
+
+var customAutoScaleAxisY: Chartist.IChartistAutoScaleAxis<Chartist.ChartistAxisUnitY> = {
+    bounds: {
+        high: 1194,
+        low: 0,
+        valueRange: 1194,
+        oom: 3,
+        step: 500,
+        min: 0,
+        max: 1500,
+        range: 1500,
+        numberOfSteps: 2,
+        values: [0, 500, 1000, 1500],
+    },
+    range: { min: 0, max: 1500 },
+    units: { pos: 'y', len: 'height', dir: 'vertical', rectStart: 'y2', rectEnd: 'y1', rectOffset: 'x1' },
+    counterUnits: { pos: 'x', len: 'width', dir: 'horizontal', rectStart: 'x1', rectEnd: 'x2', rectOffset: 'y2' },
+    chartRect: {
+        padding: { top: 12, right: 16, bottom: 0, left: 0 },
+        y2: 12,
+        y1: 120,
+        x1: 40,
+        x2: 746,
+        width: () => 0,
+        height: () => 0,
+    },
+    axisLength: 108,
+    gridOffset: 40,
+    ticks: [0, 500, 1000, 1500],
+    options: {
+        offset: 40,
+        position: 'start',
+        labelOffset: { x: 0, y: 0 },
+        showLabel: true,
+        showGrid: true,
+        scaleMinSpace: 20,
+        onlyInteger: true,
+        low: 0,
+    },
+    projectValue: (a: number) => a,
+};
+
+// Draw
```

Comment:
What follows are dumps of real event data, marginally tweaked for testing purposes.